returnn/tf/util/data.py

```diff
@@ -984,12 +984,8 @@ def declare_same_as(self, other):
     if self.dyn_size is not None and other_same_base.dyn_size is not None:
       if self.dyn_size is not other_same_base.dyn_size:
         if self.batch == other_same_base.batch and self.control_flow_ctx == other_same_base.control_flow_ctx:
-          # Note: Instead of making this a warning, we could also enforce this at some point.
-          #   The user should be able to fix `extern_data` in the config such that this is correct in the first place.
-          #   Also, in addition to this warning, we might want to add some runtime check on the eq of the dyn sizes.
-          print(
-            "Warning: assuming dim tags are same with different size placeholders: %r vs %r" % (
-              self.dyn_size, other_same_base.dyn_size))
+            BehaviorVersion.require(False,("Dim tags are same with different size placeholders: %r vs %r please check external_data" % (
```

Comment:
Please use the standard code style here in the repo. See the contribution guide.